File: flickr.py
#!/usr/bin/env python

import simplejson as json

from urllib import urlencode
from urllib2 import urlopen
from urllib import urlretrieve
import os
import sys
def photo_url(photo, size='o'):
    url = "http://farm%s.static.flickr.com/%s/%s_%s_%s.jpg" % (photo['farm'],
                                                               photo['server'],
                                                               photo['id'],
                                                               photo['secret'],
                                                               size)
    return url

YQL_PUBLIC = 'http://query.yahooapis.com/v1/public/yql'

def yql_public(query):
    # escape query
    query_str = urlencode({'q': query, 'format': 'json'})

    # fetch results
    url = '%s?%s' % (YQL_PUBLIC, query_str)
    result = urlopen(url)

    # parse json and return
    return json.load(result)['query']['results']

if __name__=='__main__':

    if len(sys.argv) == 3:
        # take tags and number of images from command line
        tags = sys.argv[1]
        n = int(sys.argv[2])
    else:
        # default to 500 pictures tagged with 'vivid'
        tags = 'vivid'
        n = 500

    # grab the top-n most interesting photos tagged with 'tags'
    query = '''select * from flickr.photos.search(%d) where tags="%s" and sort="interestingness-desc" and api_key="fa506cef10772542a871ec26351defbd"''' % (n, tags)

#    query='''select * from flickr.photos.search where has_geo="true" and text="san francisco" and api_key="fa506cef10772542a871ec26351defbd"'''

    # make yql call
    print "fetching %d photos tagged with %s from flickr" % (n, tags)
    results = yql_public(query)

    # create output directory if it doesn't exist
    directory = 'flickr_%s' % tags
    if not os.path.exists(directory):
        print "creating directory %s" % directory
        os.mkdir(directory)

    # run over results
    print "downloading %d results" % len(results['photo'])
    for photo in results['photo']:
        # build url of square image
        square_url = photo_url(photo)

        # download square image
        fname = '%s/%s.jpg' % (directory, photo['id'])
        if not os.path.exists(fname):
            print square_url , "->" , fname
            urlretrieve(square_url, filename=fname)

File: flickr.php
<!DOCTYPE html>
<html>
	<head>
		<title>Flickr Search</title>
		<link href = "site/css/bootstrap.css" rel = "stylesheet">
		<link href = "site/css/bootstrap-responsive.css" rel = "stylesheet">
		<link href = "site/css/style.css" rel = "stylesheet">
		<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
		<script src="site/js/bootstrap.js"></script>
		
		<script>
			function buildURL(photo)
			{
				var size='m';
				url = "http://farm"+photo['farm']+".static.flickr.com/"+photo['server']+"/"+photo['id']+"_"+photo['secret']+"_"+size+".jpg";
				return url
			}
			function jq(obj)
			{
				
					var container = document.getElementById("photos");
					//json="http://query.yahooapis.com/v1/public/yql?q=select%20*%20from%20flickr.photos.search%2810%29%20where%20tags%3D%22dog%22%20and%20sort%3D%22interestingness-desc%22%20and%20api_key%3D%22fa506cef10772542a871ec26351defbd%22&format=json&diagnostics=true&callback="
					for (var i=0;i<10;i++)
					{
						var img = new Image();
     					img.src = buildURL(obj.query.results.photo[i]);
						container.appendChild(img);
					}
			}
		</script>
		</title>
	</head>
	<body>
		<div class="form-group centered" style="margin-top:70px" action="" method="POST">
				<input name="query" type="text" class="form-control" placeholder="query">
				<button>Search</button>
		</div>
		<div class="centered" style="margin-top:30px" id="search">
			<button onclick="jq()">Try it</button>
		</div>
		<div id="photos">
		</div>
	<div id="carousel-example-generic" class="carousel slide">
	  <!-- Indicators -->
	  <ol class="carousel-indicators">
		<li data-target="#carousel-example-generic" data-slide-to="0" class="active"></li>
		<li data-target="#carousel-example-generic" data-slide-to="1"></li>
		<li data-target="#carousel-example-generic" data-slide-to="2"></li>
	  </ol>

	  <!-- Wrapper for slides -->
	  <div class="carousel-inner">
		<div class="item active">
		  <div class='inner-item'>
		  <img src="a.jpg" alt="">
		  </div>
		  <div class="carousel-caption">
		    slide 1
		  </div>
		</div>
	   <div class="item">
		<div class='inner-item'>
		  <img src="b.jpg"alt="">
		</div>
		  <div class="carousel-caption">
		    slide 2
		  </div>
		</div>
	   <div class="item">
		<div class='inner-item'>
		  <img src="c.jpg" alt="">
		 </div>
		  <div class="carousel-caption">
		    slide 3
		  </div>
		</div>
	  </div>

	  <!-- Controls -->
	  <a class="left carousel-control" href="#carousel-example-generic" data-slide="prev">
		<span class="icon-prev"></span>
	  </a>
	  <a class="right carousel-control" href="#carousel-example-generic" data-slide="next">
		<span class="icon-next"></span>
	  </a>
	</div>

<!--		<script src="http://query.yahooapis.com/v1/public/yql?q=select%20*%20from%20flickr.photos.search%2810%29%20where%20tags%3D%22dog%22%20and%20sort%3D%22interestingness-desc%22%20and%20api_key%3D%22fa506cef10772542a871ec26351defbd%22&format=json&diagnostics=true&callback=jq"></script>
	--></body>
</html>
